### PR TITLE
[MOBILE-1890] Add a way to mark message center messages read and deleted

### PR DESCRIPTION
**Pull request by Apekka**
### What do these changes do?
These changes add methods "MarkAsRead" and "DeleteMessage" to Xamarin NETStandard

### Why are these changes necessary?
Because customers need this

### How did you verify these changes?
By adding a button "mark as read" and "delete" in the sample app, and checking that it's working well.

### CODE REVIEW

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs [hunk 1]
```diff
@@ -216,6 +216,16 @@ public void DisplayMessage(string messageId)
             MessageCenterClass.Shared().ShowMessageCenter(messageId);
         }
 
+        public void MarkMessageRead(string messageId)
+        {
+            MessageCenterClass.Shared().Inbox.GetMessage(messageId).MarkRead();
```
Comment (rlepinski) [inline src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/Airship.cs:221]: I think you will face NPE if the message is not available. Try:
`MessageCenterClass.Shared().Inbox.MarkMessagesRead/DeleteMessages`

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs [hunk 2]
```diff
@@ -244,6 +244,34 @@ public void DisplayMessage(string messageId)
...
+        {
+            var toDelete = new UAInboxMessage[1];
+            var messages = UAMessageCenter.Shared().MessageList.Messages;
+            foreach (var message in messages)
+            {
+                if (message.MessageID == messageId)
+                {
+                    toDelete[0] = message;
+                    UAMessageCenter.Shared().MessageList.MarkMessagesDeleted(toDelete, null);
+                    Console.WriteLine("Message deleted !");
```
Comment (rlepinski) [inline src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs:270]: probably want to remove the logs

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs [hunk 3]
```diff
@@ -244,6 +244,34 @@ public void DisplayMessage(string messageId)
...
+                    message.MarkMessageRead(null);
+                    Console.WriteLine("Message read !");
+                }
+            }
+        }
+
+        public void DeleteMessage(string messageId)
+        {
+            var toDelete = new UAInboxMessage[1];
+            var messages = UAMessageCenter.Shared().MessageList.Messages;
```
Comment (rlepinski) [inline src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs:263]: Should be able to get the message with messageForID method

Comment (Apekka) [reply src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs:262]: Looks like the method messageForID is not exposed, I found it in ApiDefinitions.cs (cf this screenshot), but couldn't retrieve it in the code, I don't know why
image

Comment (marc-scig) [reply src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs:262]: `var message = UAMessageCenter.Shared().MessageList.Message(messageId);`

**Review by marc-scig — COMMENTED** [review]
I think the iOS implementation can be simplified. See my comment on DeleteMessage

**Review by Ulrico972 — APPROVED** [review]

File: src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs [hunk 4]
```diff
@@ -244,6 +244,25 @@ public void DisplayMessage(string messageId)
             UAMessageCenter.Shared().DisplayMessage(messageId);
         }
 
+        public void MarkMessageRead(string messageId)
+        {
+            var messages = UAMessageCenter.Shared().MessageList.Messages;
```
Comment (rlepinski) [inline src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs:249]: i believe you can do the same thing as you are doing for delete

**Review by marc-scig — APPROVED** [review]

**Pull request merged**